DragonLens.cs

```diff
@@ -37,5 +38,43 @@ public override void HandlePacket(BinaryReader reader, int whoAmI)
 			if (type == "PlayerManager")
 				PlayerManagerNetHandler.HandlePacket(reader);
 		}
+
+
+		public override object Call(params object[] args)
+		{
+			const string Tag = "AddAPITool";
+
+			try
+			{
+				// ── Validate signature ──────────────────────────────────────────────
+				if (args is not { Length: 4 } ||
+					args[0] is not string tag ||
+					!tag.Equals(Tag, StringComparison.Ordinal))
+				{
+					throw new ArgumentException(
+						$"{Tag} Error: Expected (\"{Tag}\", string name, Asset<Texture2D> icon, Action click).");
+				}
+
+				// ── Safe‑cast with fall‑backs ───────────────────────────────────────
+				string name = args[1] as string ?? "Unnamed";

```

Comment:
These should throw instead of falling back, improper registration should not be failing gracefully